process/sync/baseSync.go

```diff
@@ -823,6 +846,133 @@ func (boot *baseBootstrap) syncBlock() error {
 	return nil
 }
 
+// syncBlockV3 method actually does the synchronization. It requests the next block header from the pool
+// and if it is not found there it will be requested from the network. After the header is received,
+// it requests the block body in the same way(pool and then, if it is not found in the pool, from network).
+// Once received, the header is verified through VerifyBlockProposal, but not before warming up the tx pool.
+// Finally, if everything works, the block will be committed and added into the processing queue.
+// And all this mechanism will be reiterated for the next block.
+func (boot *baseBootstrap) syncBlockV3() error {
+	boot.computeNodeState()
+	nodeState := boot.GetNodeState()
+	if nodeState != common.NsNotSynchronized {
+		return nil
+	}
+
+	defer func() {
+		boot.mutNodeState.Lock()
+		boot.isNodeStateCalculated = false
+		boot.mutNodeState.Unlock()
+	}()
+
+	if boot.forkInfo.IsDetected {
+		// should be no forks after Andromeda
+		log.Error("fork detected",
+			"nonce", boot.forkInfo.Nonce,
+			"hash", boot.forkInfo.Hash,
+		)
+
+		return ErrForkDetected
+	}
+
+	var body data.BodyHandler
+	var header data.HeaderHandler
+	var err error
+
+	defer func() {
+		if err != nil {
+			log.Warn("sync block failed", "error", err)
+
+			boot.doJobOnSyncBlockFail(body, header, err)
+		}
+	}()
+
+	header, err = boot.getNextHeaderRequestingIfMissing()
+	if err != nil {
+		return err
+	}
+
+	go boot.requestHeadersFromNonceIfMissing(header.GetNonce() + 1)
```

Comment:
should we consider modifying `MaxHeadersToRequestInAdvance` here? we might need to request more headers in advance